Framework/src/MonitorObjectCollection.cxx

```diff
@@ -46,16 +45,19 @@ void MonitorObjectCollection::merge(mergers::MergeInterface* const other)
       if (!otherMO || !targetMO) {
         throw std::runtime_error("The target object or the other object could not be casted to MonitorObject.");
       }
-      if (!reportedMismatchingRunNumbers && targetMO->getActivity().mId != otherMO->getActivity().mId) {
+
+      if (otherMO->getActivity().mId > targetMO->getActivity().mId) {
```

Comment:
yes, but we should still produce a warning and ignore an input object if it's run number is lower, and as before, avoid flooding the IL for the rest of `otherCollection`. In this version of the code, you would simply merge such input anyway.